internal/app/admin/http/handler.go

```diff
@@ -0,0 +1,136 @@
+package handler
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	v2 "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	"github.com/envoyproxy/xds-relay/internal/app/cache"
+
+	"github.com/envoyproxy/xds-relay/internal/app/orchestrator"
+
+	bootstrapv1 "github.com/envoyproxy/xds-relay/pkg/api/bootstrap/v1"
+)
+
+type Handler struct {
+	prefix      string
+	description string
+	handler     http.HandlerFunc
+}
+
+func getHandlers(bootstrap *bootstrapv1.Bootstrap, orchestrator *orchestrator.Orchestrator) []Handler {
+	handlers := []Handler{
+		{
+			"/",
+			"admin home page",
+			defaultHandler(nil),
+		},
+		{
+			"/cache/",
+			"print cache entry for a given key. usage: `/cache/<key>`",
+			cacheDumpHandler(orchestrator),
+		},
+		{
+			"/server_info",
+			"print bootstrap configuration",
+			configDumpHandler(bootstrap),
+		},
+	}
+	// The default handler is defined later to avoid infinite recursion.
+	handlers[0].handler = defaultHandler(handlers)
+	return handlers
+}
+
+func RegisterHandlers(bootstrapConfig *bootstrapv1.Bootstrap, orchestrator *orchestrator.Orchestrator) {
+	for _, handler := range getHandlers(bootstrapConfig, orchestrator) {
+		http.Handle(handler.prefix, handler.handler)
+	}
+}
+
+func defaultHandler(handlers []Handler) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		// The "/" pattern matches everything, so we need to check
+		// that we're at the root here.
+		if req.URL.Path != "/" {
+			http.NotFound(w, req)
+			return
+		}
+		fmt.Fprintf(w, "admin commands are:\n")
+		for _, handler := range handlers {
+			fmt.Fprintf(w, "  %s: %s\n", handler.prefix, handler.description)
+		}
+	}
+}
+
+func configDumpHandler(bootstrapConfig *bootstrapv1.Bootstrap) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		configString, err := interfaceToString(bootstrapConfig)
+		if err != nil {
+			fmt.Fprintf(w, "Failed to dump config: %s\n", err.Error())
+		}
+		fmt.Fprintf(w, "%s\n", configString)
+	}
+}
+
+// TODO(lisalu): Support dump of entire cache when no key is provided.
+// TODO(lisalu): Support dump of matching resources when cache key regex is provided.
```

Comment:
It may make sense to add issues for these and follow up in a separate PR.

Reply:
opened #88 and #89 separately from #75